Not resolving dependancy

Hey 

I have had an issue with our grade build file.  I have recreated it with the following example, using gradle 4.5.1:

```groovy
plugins {
    id 'com.github.blindpirate.gogradle' version '0.9.1'
}

golang {
    packagePath = 'github.com/luceracloud/tmp'
    goVersion = '1.10.1'
}

dependencies {
    golang {
        build(name: 'github.com/golang/protobuf', tag: 'v1.1.0')
    }
}

build {
    targetPlatform = ['darwin-amd64', 'freebsd-amd64']
    outputLocation = './.gogradle/build/${GOOS}_${GOARCH}/usr/local/bin/${PROJECT_NAME}${GOEXE}'
}
```

Running `gradle vendor` I see the following issue:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':resolveBuildDependencies'.
> Exception in resolution, message is:
  Cannot resolve dependency:git.apache.org/thrift.git: commit='88591e32e710a0524327153c8b629d5b461e35e0', branch='master', urls=[github.com/apache/thrift], subpackages='[lib/go/thrift]'
  Resolution stack is:
  +- github.com/luceracloud/tmp
   +- github.com/golang/protobuf#b4deda0973fb4c70b50d226b1af49f3da59f5265
    +- google.golang.org/genproto#694d95ba50e67b2e363f3483057db5d4910c18f9
     +- google.golang.org/grpc#1fa3750c9a209f5efbdc2722d40f2657885af5e5
      +- golang.org/x/oauth2#770e5ebd4ab27c4d70b0983674d138217f5e2c72
       +- cloud.google.com/go#c338ed132971aadcd5157ddef26adf1c9bed7523
        +- go.opencensus.io#a6584076435b66d4a8ee67307e193e81b583f705
```

http://github.com/apache/thrift is a valid repo and `lib/go/thrift` is the valid sub package for the go files so I am trying to figure out why this isn't able to resolve the dependancy.  

Using the following works ok:  `go get github.com/apache/thrift/lib/go/thrift`

Is this a dependancy thats not supported due to the sub-package?

Thanks for your help

Phil

Full trace:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':resolveBuildDependencies'.
> Exception in resolution, message is:
  Cannot resolve dependency:git.apache.org/thrift.git: commit='88591e32e710a0524327153c8b629d5b461e35e0', branch='master', urls=[github.com/apache/thrift], subpackages='[lib/go/thrift]'
  Resolution stack is:
  +- github.com/luceracloud/tmp
   +- github.com/golang/protobuf#b4deda0973fb4c70b50d226b1af49f3da59f5265
    +- google.golang.org/genproto#694d95ba50e67b2e363f3483057db5d4910c18f9
     +- google.golang.org/grpc#1fa3750c9a209f5efbdc2722d40f2657885af5e5
      +- golang.org/x/oauth2#770e5ebd4ab27c4d70b0983674d138217f5e2c72
       +- cloud.google.com/go#c338ed132971aadcd5157ddef26adf1c9bed7523
        +- go.opencensus.io#a6584076435b66d4a8ee67307e193e81b583f705


* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':resolveBuildDependencies'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:623)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:578)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: com.github.blindpirate.gogradle.core.exceptions.ResolutionStackWrappingException: Exception in resolution, message is:
Cannot resolve dependency:git.apache.org/thrift.git: commit='88591e32e710a0524327153c8b629d5b461e35e0', branch='master', urls=[github.com/apache/thrift], subpackages='[lib/go/thrift]'
Resolution stack is:
+- github.com/luceracloud/tmp
 +- github.com/golang/protobuf#b4deda0973fb4c70b50d226b1af49f3da59f5265
  +- google.golang.org/genproto#694d95ba50e67b2e363f3483057db5d4910c18f9
   +- google.golang.org/grpc#1fa3750c9a209f5efbdc2722d40f2657885af5e5
    +- golang.org/x/oauth2#770e5ebd4ab27c4d70b0983674d138217f5e2c72
     +- cloud.google.com/go#c338ed132971aadcd5157ddef26adf1c9bed7523
      +- go.opencensus.io#a6584076435b66d4a8ee67307e193e81b583f705

        at com.github.blindpirate.gogradle.core.exceptions.ResolutionStackWrappingException.wrapWithResolutionStack(ResolutionStackWrappingException.java:51)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.resolve(DependencyTreeFactory.java:79)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.getTree(DependencyTreeFactory.java:39)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolveDependencies(ResolveDependencies.java:172)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolve(ResolveDependencies.java:153)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 30 more
Caused by: com.github.blindpirate.gogradle.core.exceptions.DependencyResolutionException: Cannot resolve dependency:git.apache.org/thrift.git: commit='88591e32e710a0524327153c8b629d5b461e35e0', branch='master', urls=[github.com/apache/thrift], subpackages='[lib/go/thrift]'
        at com.github.blindpirate.gogradle.core.exceptions.DependencyResolutionException.cannotResolveDependency(DependencyResolutionException.java:56)
        at com.github.blindpirate.gogradle.core.dependency.resolve.AbstractVcsDependencyManager.doResolve(AbstractVcsDependencyManager.java:73)
        at com.github.blindpirate.gogradle.core.dependency.resolve.CacheEnabledDependencyResolverMixin.lambda$resolve$0(CacheEnabledDependencyResolverMixin.java:38)
        at com.github.blindpirate.gogradle.core.cache.CloneBackedCache.get(CloneBackedCache.java:34)
        at com.github.blindpirate.gogradle.core.cache.ProjectCacheManager.resolve(ProjectCacheManager.java:59)
        at com.github.blindpirate.gogradle.core.dependency.resolve.CacheEnabledDependencyResolverMixin.resolve(CacheEnabledDependencyResolverMixin.java:36)
        at com.github.blindpirate.gogradle.vcs.git.GitNotationDependency.doResolve(GitNotationDependency.java:28)
        at com.github.blindpirate.gogradle.core.dependency.AbstractNotationDependency.resolve(AbstractNotationDependency.java:55)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.createSubContextAndResolve(DependencyTreeFactory.java:88)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.lambda$resolve$0(DependencyTreeFactory.java:76)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.resolve(DependencyTreeFactory.java:77)
        ... 46 more
Caused by: com.github.blindpirate.gogradle.core.exceptions.DependencyResolutionException: Cloning repository of git.apache.org/thrift.git failed.
        at com.github.blindpirate.gogradle.core.exceptions.DependencyResolutionException.cannotCloneRepository(DependencyResolutionException.java:39)
        at com.github.blindpirate.gogradle.vcs.GitMercurialDependencyManager.tryCloneWithUrls(GitMercurialDependencyManager.java:190)
        at com.github.blindpirate.gogradle.vcs.GitMercurialDependencyManager.initRepository(GitMercurialDependencyManager.java:173)
        at com.github.blindpirate.gogradle.core.dependency.resolve.AbstractVcsDependencyManager.resolveRepository(AbstractVcsDependencyManager.java:117)
        at com.github.blindpirate.gogradle.core.dependency.resolve.AbstractVcsDependencyManager.doResolve(AbstractVcsDependencyManager.java:63)
        ... 55 more
Caused by: com.github.blindpirate.gogradle.core.exceptions.BuildException: Build failed due to exception of cmd: git clone --recursive --progress github.com/apache/thrift /Users/ppreston/.gradle/go/repo/git.apache.org/thrift.git/a10a95de1fc4d5545f5706e022335c5c
in: /Users/ppreston/Developer/Projects/Golang/src/github.com/luceracloud/tmp/.
with env:
 PATH=/Users/ppreston/.sdkman/candidates/springboot/current/bin:/Users/ppreston/.sdkman/candidates/maven/current/bin:/Users/ppreston/.sdkman/candidates/java/current/bin:/Users/ppreston/.sdkman/candidates/groovyserv/current/bin:/Users/ppreston/.sdkman/candidates/groovy/current/bin:/Users/ppreston/.sdkman/candidates/grails/current/bin:/Users/ppreston/.sdkman/candidates/gradle/current/bin:/Users/ppreston/.sdkman/candidates/ant/current/bin:/Users/ppreston/.rbenv/shims:/Users/ppreston/Developer/Projects/Golang/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
exception:com.github.blindpirate.gogradle.core.exceptions.BuildException: Build failed due to return code 128 of cmd: git clone --recursive --progress github.com/apache/thrift /Users/ppreston/.gradle/go/repo/git.apache.org/thrift.git/a10a95de1fc4d5545f5706e022335c5c
in: /Users/ppreston/Developer/Projects/Golang/src/github.com/luceracloud/tmp/.
stderr: fatal: repository 'github.com/apache/thrift' does not exist
stdout:
with env:
 PATH=/Users/ppreston/.sdkman/candidates/springboot/current/bin:/Users/ppreston/.sdkman/candidates/maven/current/bin:/Users/ppreston/.sdkman/candidates/java/current/bin:/Users/ppreston/.sdkman/candidates/groovyserv/current/bin:/Users/ppreston/.sdkman/candidates/groovy/current/bin:/Users/ppreston/.sdkman/candidates/grails/current/bin:/Users/ppreston/.sdkman/candidates/gradle/current/bin:/Users/ppreston/.sdkman/candidates/ant/current/bin:/Users/ppreston/.rbenv/shims:/Users/ppreston/Developer/Projects/Golang/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
        at com.github.blindpirate.gogradle.core.exceptions.BuildException.processInteractionFailed(BuildException.java:82)
        at com.github.blindpirate.gogradle.vcs.GitMercurialAccessor.runWithProgress(GitMercurialAccessor.java:137)
        at com.github.blindpirate.gogradle.vcs.GitMercurialAccessor.runWithProgress(GitMercurialAccessor.java:112)
        at com.github.blindpirate.gogradle.vcs.git.GitClientAccessor.clone(GitClientAccessor.java:205)
        at com.github.blindpirate.gogradle.vcs.GitMercurialDependencyManager.tryCloneWithUrls(GitMercurialDependencyManager.java:185)
        at com.github.blindpirate.gogradle.vcs.GitMercurialDependencyManager.initRepository(GitMercurialDependencyManager.java:173)
        at com.github.blindpirate.gogradle.core.dependency.resolve.AbstractVcsDependencyManager.resolveRepository(AbstractVcsDependencyManager.java:117)
        at com.github.blindpirate.gogradle.core.dependency.resolve.AbstractVcsDependencyManager.doResolve(AbstractVcsDependencyManager.java:63)
        at com.github.blindpirate.gogradle.core.dependency.resolve.CacheEnabledDependencyResolverMixin.lambda$resolve$0(CacheEnabledDependencyResolverMixin.java:38)
        at com.github.blindpirate.gogradle.core.cache.CloneBackedCache.get(CloneBackedCache.java:34)
        at com.github.blindpirate.gogradle.core.cache.ProjectCacheManager.resolve(ProjectCacheManager.java:59)
        at com.github.blindpirate.gogradle.core.dependency.resolve.CacheEnabledDependencyResolverMixin.resolve(CacheEnabledDependencyResolverMixin.java:36)
        at com.github.blindpirate.gogradle.vcs.git.GitNotationDependency.doResolve(GitNotationDependency.java:28)
        at com.github.blindpirate.gogradle.core.dependency.AbstractNotationDependency.resolve(AbstractNotationDependency.java:55)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.createSubContextAndResolve(DependencyTreeFactory.java:88)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.lambda$resolve$0(DependencyTreeFactory.java:76)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.resolve(DependencyTreeFactory.java:77)
        at com.github.blindpirate.gogradle.core.dependency.tree.DependencyTreeFactory.getTree(DependencyTreeFactory.java:39)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolveDependencies(ResolveDependencies.java:172)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolve(ResolveDependencies.java:153)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:623)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:578)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.lang.Thread.run(Thread.java:748)

        at com.github.blindpirate.gogradle.core.exceptions.BuildException.processInteractionFailed(BuildException.java:96)
        at com.github.blindpirate.gogradle.vcs.GitMercurialAccessor.runWithProgress(GitMercurialAccessor.java:145)
        at com.github.blindpirate.gogradle.vcs.GitMercurialAccessor.runWithProgress(GitMercurialAccessor.java:112)
        at com.github.blindpirate.gogradle.vcs.git.GitClientAccessor.clone(GitClientAccessor.java:205)
        at com.github.blindpirate.gogradle.vcs.GitMercurialDependencyManager.tryCloneWithUrls(GitMercurialDependencyManager.java:185)
        ... 58 more


* Get more help at https://help.gradle.org

BUILD FAILED in 15s
2 actionable tasks: 2 executed```

--- a/src/integration-test/groovy/com/github/blindpirate/gogradle/issues/RecognizeGopkgLockIntegrationTest.groovy
+++ b/src/integration-test/groovy/com/github/blindpirate/gogradle/issues/RecognizeGopkgLockIntegrationTest.groovy
@@ -1,0 +1,40 @@
+package com.github.blindpirate.gogradle.issues
+
+import com.github.blindpirate.gogradle.GogradleRunner
+import com.github.blindpirate.gogradle.support.AccessWeb
+import com.github.blindpirate.gogradle.support.IntegrationTestSupport
+import com.github.blindpirate.gogradle.support.WithMockGo
+import com.github.blindpirate.gogradle.support.WithResource
+import com.github.blindpirate.gogradle.util.IOUtils
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@WithResource
+@RunWith(GogradleRunner)
+@WithMockGo
+@AccessWeb
+class RecognizeGopkgLockIntegrationTest extends IntegrationTestSupport {
+    // https://github.com/gogradle/gogradle/issues/221
+    // https://github.com/census-instrumentation/opencensus-go/blob/0edc045e110a4ba034ed03dffbbaf13eeae8b25b/Gopkg.lock
+    @Test
+    void "can recognize Gopkg.lock's source"() {
+        writeBuildAndSettingsDotGradle(buildDotGradleBase)
+        IOUtils.write(resource, 'Gopkg.lock', '''
+[[projects]]
+  branch = "master"
+  name = "git.apache.org/thrift.git"
+  packages = ["lib/go/thrift"]
+  revision = "606f1ef31447526b908244933d5b716397a6bad8"
+  source = "github.com/apache/thrift"
+''')
+        newBuild('init')
+        newBuild('vendor')
+
+        assert new File(resource, 'vendor/git.apache.org/thrift.git').exists()
+    }
+
+    @Override
+    File getProjectRoot() {
+        return resource
+    }
+}

--- a/src/main/java/com/github/blindpirate/gogradle/GogradleModule.java
+++ b/src/main/java/com/github/blindpirate/gogradle/GogradleModule.java
@@ -61,7 +61,6 @@ import com.github.blindpirate.gogradle.core.pack.UnrecognizedPackagePathResolver
 import com.github.blindpirate.gogradle.core.pack.VcsPackagePathResolver;
 import com.github.blindpirate.gogradle.crossplatform.DefaultGoBinaryManager;
 import com.github.blindpirate.gogradle.crossplatform.GoBinaryManager;
-import com.github.blindpirate.gogradle.util.CollectionUtils;
 import com.github.blindpirate.gogradle.util.logging.DebugLog;
 import com.github.blindpirate.gogradle.util.logging.DebugLogMethodInterceptor;
 import com.github.blindpirate.gogradle.vcs.Bazaar;
@@ -95,6 +94,8 @@ import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import static com.github.blindpirate.gogradle.util.CollectionUtils.immutableList;
+
 /**
  * Provides configurations for Guice dependency injection.
  */
@@ -123,7 +124,7 @@ public class GogradleModule extends AbstractModule {
         bind(MapNotationParser.class).to(DefaultMapNotationParser.class);
         bind(GlobalCacheManager.class).to(DefaultGlobalCacheManager.class);
         bind(DependencyRegistry.class).to(DefaultDependencyRegistry.class);
-        bind(PackagePathResolver.class).to(DefaultPackagePathResolver.class);
+//        bind(PackagePathResolver.class).to(DefaultPackagePathResolver.class);
         bind(NotationConverter.class).to(DefaultNotationConverter.class);
         bind(BuildConstraintManager.class).to(DefaultBuildConstraintManager.class);
         bind(DependencyVisitor.class).to(DefaultDependencyVisitor.class);
@@ -172,7 +173,7 @@ public class GogradleModule extends AbstractModule {
             GlockDependencyFactory glockDependencyFactory,
             GopmDependencyFactory gopmDependencyFactory,
             GpmDependencyFactory gpmDependencyFactory) {
-        return CollectionUtils.immutableList(
+        return immutableList(
                 lockedDependencyManager,
                 depDependencyFactory,
                 godepDependencyFactory,
@@ -186,11 +187,15 @@ public class GogradleModule extends AbstractModule {
 
     }
 
+    /**
+     * PackagePathResolver which supports package substitution.
+     * See https://github.com/gogradle/gogradle/blob/master/docs/repository-management.md
+     */
     @Inject
     @Provides
     @Singleton
-    @DefaultPackagePathResolver.PackagePathResolvers
-    public List<PackagePathResolver> packagePathResolvers(
+    @DefaultPackagePathResolver.AllPackagePathResolvers
+    public PackagePathResolver allPackagePathResolver(
             RepositoryHandlerPathResolver repositoryHandlerPathResolver,
             BitbucketPackagePathResolver bitbucketPackagePathResolver,
             IBMDevOpsPackagePathResolver ibmDevOpsPackagePathResolver,
@@ -199,7 +204,7 @@ public class GogradleModule extends AbstractModule {
             VcsPackagePathResolver vcsPackagePathResolver,
             MetadataPackagePathResolver metadataPackagePathResolver,
             UnrecognizedPackagePathResolver unrecognizedPackagePathResolver) {
-        return CollectionUtils.immutableList(
+        return new DefaultPackagePathResolver(
                 repositoryHandlerPathResolver,
                 standardPackagePathResolver,
                 new GithubGitlabPackagePathResolver("github.com"),
@@ -213,5 +218,32 @@ public class GogradleModule extends AbstractModule {
         );
     }
 
-
+    /**
+     * PackagePathResolver which doesn't support package substitution.
+     * E.g. only produces "original" packages (specified by https://golang.org/cmd/go/#hdr-Remote_import_paths).
+     */
+    @Inject
+    @Provides
+    @Singleton
+    @DefaultPackagePathResolver.OriginalPackagePathResolvers
+    public PackagePathResolver originalPackagePathResolver(
+            BitbucketPackagePathResolver bitbucketPackagePathResolver,
+            IBMDevOpsPackagePathResolver ibmDevOpsPackagePathResolver,
+            StandardPackagePathResolver standardPackagePathResolver,
+            GlobalCachePackagePathResolver globalCachePackagePathResolver,
+            VcsPackagePathResolver vcsPackagePathResolver,
+            MetadataPackagePathResolver metadataPackagePathResolver,
+            UnrecognizedPackagePathResolver unrecognizedPackagePathResolver) {
+        return new DefaultPackagePathResolver(
+                standardPackagePathResolver,
+                new GithubGitlabPackagePathResolver("github.com"),
+                new GithubGitlabPackagePathResolver("gitlab.com"),
+                bitbucketPackagePathResolver,
+                ibmDevOpsPackagePathResolver,
+                globalCachePackagePathResolver,
+                vcsPackagePathResolver,
+                metadataPackagePathResolver,
+                unrecognizedPackagePathResolver
+        );
+    }
 }

--- a/src/main/java/com/github/blindpirate/gogradle/core/cache/PersistenceNotationToResolvedCache.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/cache/PersistenceNotationToResolvedCache.java
@@ -31,6 +31,7 @@ import java.util.Map;
 import java.util.stream.Collectors;
 
 import static com.github.blindpirate.gogradle.GogradleGlobal.GOGRADLE_COMPATIBLE_VERSION;
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
 
 @Singleton
 public class PersistenceNotationToResolvedCache
@@ -38,7 +39,8 @@ public class PersistenceNotationToResolvedCache
     private final PackagePathResolver packagePathResolver;
 
     @Inject
-    public PersistenceNotationToResolvedCache(Project project, PackagePathResolver packagePathResolver) {
+    public PersistenceNotationToResolvedCache(Project project,
+                                              @AllPackagePathResolvers PackagePathResolver packagePathResolver) {
         super(project, "PersistenceNotationToResolvedCache-" + GOGRADLE_COMPATIBLE_VERSION + ".bin");
         this.packagePathResolver = packagePathResolver;
     }

--- a/src/main/java/com/github/blindpirate/gogradle/core/cache/PersistenceResolvedToDependenciesCache.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/cache/PersistenceResolvedToDependenciesCache.java
@@ -31,6 +31,7 @@ import java.util.Map;
 import java.util.stream.Collectors;
 
 import static com.github.blindpirate.gogradle.GogradleGlobal.GOGRADLE_COMPATIBLE_VERSION;
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
 
 
 @Singleton
@@ -40,7 +41,8 @@ public class PersistenceResolvedToDependenciesCache
     private final PackagePathResolver packagePathResolver;
 
     @Inject
-    public PersistenceResolvedToDependenciesCache(Project project, PackagePathResolver packagePathResolver) {
+    public PersistenceResolvedToDependenciesCache(Project project,
+                                                  @AllPackagePathResolvers PackagePathResolver packagePathResolver) {
         super(project, "PersistenceResolvedToDependenciesCache-"
                 + GOGRADLE_COMPATIBLE_VERSION + ".bin");
         this.packagePathResolver = packagePathResolver;

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/parse/DefaultMapNotationParser.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/parse/DefaultMapNotationParser.java
@@ -28,6 +28,7 @@ import com.github.blindpirate.gogradle.core.dependency.NotationDependency;
 import com.github.blindpirate.gogradle.core.dependency.UnrecognizedNotationDependency;
 import com.github.blindpirate.gogradle.core.exceptions.DependencyResolutionException;
 import com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver;
+import com.github.blindpirate.gogradle.core.pack.PackagePathResolver;
 import com.github.blindpirate.gogradle.util.Assert;
 import com.github.blindpirate.gogradle.util.MapUtils;
 import com.github.blindpirate.gogradle.util.StringUtils;
@@ -38,6 +39,7 @@ import javax.inject.Singleton;
 import java.util.List;
 import java.util.Map;
 
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
 import static com.github.blindpirate.gogradle.util.MapUtils.getString;
 import static com.github.blindpirate.gogradle.vcs.VcsNotationDependency.URLS_KEY;
 import static com.github.blindpirate.gogradle.vcs.VcsNotationDependency.URL_KEY;
@@ -56,10 +58,10 @@ public class DefaultMapNotationParser implements MapNotationParser {
     @Inject
     public DefaultMapNotationParser(DirMapNotationParser dirMapNotationParser,
                                     VendorMapNotationParser vendorMapNotationParser,
-                                    DefaultPackagePathResolver packagePathResolver) {
+                                    @AllPackagePathResolvers PackagePathResolver packagePathResolver) {
         this.dirMapNotationParser = dirMapNotationParser;
         this.vendorMapNotationParser = vendorMapNotationParser;
-        this.packagePathResolver = packagePathResolver;
+        this.packagePathResolver = (DefaultPackagePathResolver) packagePathResolver;
     }
 
     @Override

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/parse/DefaultNotationConverter.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/parse/DefaultNotationConverter.java
@@ -28,12 +28,14 @@ import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.util.Map;
 
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
+
 @Singleton
 public class DefaultNotationConverter implements NotationConverter {
     private final PackagePathResolver packagePathResolver;
 
     @Inject
-    public DefaultNotationConverter(PackagePathResolver packagePathResolver) {
+    public DefaultNotationConverter(@AllPackagePathResolvers PackagePathResolver packagePathResolver) {
         this.packagePathResolver = packagePathResolver;
     }
 

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/ExternalDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/ExternalDependencyFactory.java
@@ -37,6 +37,7 @@ import java.util.stream.Collectors;
 
 import static com.github.blindpirate.gogradle.core.GolangConfiguration.BUILD;
 import static com.github.blindpirate.gogradle.core.GolangConfiguration.TEST;
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
 import static com.github.blindpirate.gogradle.util.DependencySetUtils.parseMany;
 import static com.github.blindpirate.gogradle.util.StringUtils.pathStartsWith;
 import static com.google.common.collect.ImmutableMap.of;
@@ -48,6 +49,7 @@ public abstract class ExternalDependencyFactory {
     protected MapNotationParser mapNotationParser;
 
     @Inject
+    @AllPackagePathResolvers
     protected PackagePathResolver packagePathResolver;
 
     @Inject

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/FirstPassVendorDirectoryVisitor.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/FirstPassVendorDirectoryVisitor.java
@@ -38,8 +38,7 @@ public class FirstPassVendorDirectoryVisitor extends SimpleFileVisitor<Path> {
 
     private final Path parentVendor;
 
-    public FirstPassVendorDirectoryVisitor(Path parentVendor,
-                                           PackagePathResolver packagePathResolver) {
+    public FirstPassVendorDirectoryVisitor(Path parentVendor, PackagePathResolver packagePathResolver) {
         this.packagePathResolver = packagePathResolver;
         this.parentVendor = parentVendor;
     }

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/SourceCodeDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/SourceCodeDependencyFactory.java
@@ -27,6 +27,7 @@ import com.github.blindpirate.gogradle.core.dependency.GolangDependency;
 import com.github.blindpirate.gogradle.core.dependency.GolangDependencySet;
 import com.github.blindpirate.gogradle.core.dependency.ResolvedDependency;
 import com.github.blindpirate.gogradle.core.dependency.parse.NotationParser;
+import com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver;
 import com.github.blindpirate.gogradle.core.pack.PackagePathResolver;
 
 import javax.inject.Inject;
@@ -36,6 +37,7 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.*;
 import static com.github.blindpirate.gogradle.util.StringUtils.isBlank;
 
 /**
@@ -52,10 +54,11 @@ public class SourceCodeDependencyFactory {
     private final GogradleRootProject gogradleRootProject;
 
     @Inject
-    public SourceCodeDependencyFactory(PackagePathResolver packagePathResolver,
-                                       NotationParser notationParser,
-                                       GoImportExtractor goImportExtractor,
-                                       GogradleRootProject gogradleRootProject) {
+    public SourceCodeDependencyFactory(
+            @AllPackagePathResolvers PackagePathResolver packagePathResolver,
+            NotationParser notationParser,
+            GoImportExtractor goImportExtractor,
+            GogradleRootProject gogradleRootProject) {
         this.packagePathResolver = packagePathResolver;
         this.notationParser = notationParser;
         this.goImportExtractor = goImportExtractor;

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/SourceCodeDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/SourceCodeDependencyFactory.java
@@ -27,7 +27,6 @@ import com.github.blindpirate.gogradle.core.dependency.GolangDependency;
 import com.github.blindpirate.gogradle.core.dependency.GolangDependencySet;
 import com.github.blindpirate.gogradle.core.dependency.ResolvedDependency;
 import com.github.blindpirate.gogradle.core.dependency.parse.NotationParser;
-import com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver;
 import com.github.blindpirate.gogradle.core.pack.PackagePathResolver;
 
 import javax.inject.Inject;
@@ -37,7 +36,7 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.*;
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
 import static com.github.blindpirate.gogradle.util.StringUtils.isBlank;
 
 /**

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/VendorDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/VendorDependencyFactory.java
@@ -28,6 +28,8 @@ import javax.inject.Singleton;
 import java.io.File;
 import java.nio.file.Path;
 
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.AllPackagePathResolvers;
+
 
 /**
  * A {@link VendorDependencyFactory} is a factory that reads vendor directory and resolves them to
@@ -40,7 +42,7 @@ public class VendorDependencyFactory {
     private final PackagePathResolver packagePathResolver;
 
     @Inject
-    public VendorDependencyFactory(PackagePathResolver packagePathResolver) {
+    public VendorDependencyFactory(@AllPackagePathResolvers PackagePathResolver packagePathResolver) {
         this.packagePathResolver = packagePathResolver;
     }
 

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactory.java
@@ -32,6 +32,6 @@ public class DepDependencyFactory extends ExternalDependencyFactory {
 
     @Override
     protected List<Map<String, Object>> adapt(File file) {
-        return GopkgDotLockModel.parse(file);
+        return GopkgDotLockModel.parse(originalPackagePathResolver, file);
     }
 }

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactory.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactory.java
@@ -1,11 +1,15 @@
 package com.github.blindpirate.gogradle.core.dependency.produce.external.dep;
 
 import com.github.blindpirate.gogradle.core.dependency.produce.ExternalDependencyFactory;
+import com.github.blindpirate.gogradle.core.pack.PackagePathResolver;
 
+import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.io.File;
 import java.util.List;
 import java.util.Map;
+
+import static com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.OriginalPackagePathResolvers;
 
 /**
  * Converts Gopkg.lock in repos managed by dep to gogradle map notations.
@@ -14,6 +18,13 @@ import java.util.Map;
  */
 @Singleton
 public class DepDependencyFactory extends ExternalDependencyFactory {
+    private final PackagePathResolver originalPackagePathResolver;
+
+    @Inject
+    public DepDependencyFactory(@OriginalPackagePathResolvers PackagePathResolver originalPackagePathResolver) {
+        this.originalPackagePathResolver = originalPackagePathResolver;
+    }
+
     @Override
     public String identityFileName() {
         return "Gopkg.lock";

--- a/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/GopkgDotLockModel.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/GopkgDotLockModel.java
@@ -1,6 +1,9 @@
 package com.github.blindpirate.gogradle.core.dependency.produce.external.dep;
 
+import com.github.blindpirate.gogradle.core.VcsGolangPackage;
 import com.github.blindpirate.gogradle.core.dependency.NotationDependency;
+import com.github.blindpirate.gogradle.core.pack.PackagePathResolver;
+import com.github.blindpirate.gogradle.vcs.VcsScheme;
 import com.google.common.collect.ImmutableMap;
 import com.moandjiezana.toml.Toml;
 
@@ -9,6 +12,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * Model of Gopkg.lock file managed by dep.
@@ -19,17 +23,36 @@ public class GopkgDotLockModel {
     private static final Map<String, String> PROPERTY_NAME_CONVERSION =
             ImmutableMap.of("packages", "subpackages",
                     "revision", "commit",
-                    "version", "tag",
-                    "source", "url");
+                    "version", "tag");
     private static final Map<String, Function<List, List>> PROPERTY_CONVERSION =
             ImmutableMap.of("packages", GopkgDotLockModel::convertSubpackages);
 
-    public static List<Map<String, Object>> parse(File file) {
+    public static List<Map<String, Object>> parse(PackagePathResolver packagePathResolver, File file) {
         Toml toml = new Toml().read(file);
         List<Map<String, Object>> projects = toml.getList("projects");
         convertProperties(projects);
         convertPropertyNames(projects);
+        processSource(packagePathResolver, projects);
         return projects;
+    }
+
+    private static void processSource(PackagePathResolver packagePathResolver, List<Map<String, Object>> projects) {
+        projects.forEach(project -> {
+            String source = (String) project.remove("source");
+            if (source != null) {
+                if (startWithVcsScheme(source)) {
+                    project.put("url", source);
+                } else {
+                    VcsGolangPackage pkg = (VcsGolangPackage) packagePathResolver.produce(source).get();
+                    project.put("urls", pkg.getUrls());
+                    project.put("vcs", pkg.getVcsType().getName());
+                }
+            }
+        });
+    }
+
+    private static boolean startWithVcsScheme(String source) {
+        return Stream.of(VcsScheme.values()).map(VcsScheme::getScheme).anyMatch(source::startsWith);
     }
 
     private static void convertPropertyNames(List<Map<String, Object>> projects) {

--- a/src/main/java/com/github/blindpirate/gogradle/core/pack/DefaultPackagePathResolver.java
+++ b/src/main/java/com/github/blindpirate/gogradle/core/pack/DefaultPackagePathResolver.java
@@ -18,6 +18,7 @@
 package com.github.blindpirate.gogradle.core.pack;
 
 import com.github.blindpirate.gogradle.core.GolangPackage;
+import com.github.blindpirate.gogradle.util.CollectionUtils;
 import com.github.blindpirate.gogradle.util.FactoryUtil;
 import com.github.blindpirate.gogradle.util.logging.DebugLog;
 import com.google.inject.BindingAnnotation;
@@ -46,8 +47,8 @@ public class DefaultPackagePathResolver implements PackagePathResolver {
     private final List<PackagePathResolver> delegates;
 
     @Inject
-    public DefaultPackagePathResolver(@PackagePathResolvers List<PackagePathResolver> delegates) {
-        this.delegates = delegates;
+    public DefaultPackagePathResolver(PackagePathResolver... delegates) {
+        this.delegates = CollectionUtils.immutableList(delegates);
     }
 
     @Override
@@ -85,6 +86,12 @@ public class DefaultPackagePathResolver implements PackagePathResolver {
     @BindingAnnotation
     @Target({FIELD, PARAMETER, METHOD})
     @Retention(RUNTIME)
-    public @interface PackagePathResolvers {
+    public @interface AllPackagePathResolvers {
+    }
+
+    @BindingAnnotation
+    @Target({FIELD, PARAMETER, METHOD})
+    @Retention(RUNTIME)
+    public @interface OriginalPackagePathResolvers {
     }
 }

--- a/src/main/java/com/github/blindpirate/gogradle/vcs/VcsScheme.java
+++ b/src/main/java/com/github/blindpirate/gogradle/vcs/VcsScheme.java
@@ -18,6 +18,10 @@ public enum VcsScheme {
         this.scheme = scheme;
     }
 
+    public String getScheme() {
+        return scheme;
+    }
+
     public String buildUrl(String packagePath) {
         return scheme + packagePath;
     }

--- a/src/test/groovy/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactoryTest.groovy
+++ b/src/test/groovy/com/github/blindpirate/gogradle/core/dependency/produce/external/dep/DepDependencyFactoryTest.groovy
@@ -19,6 +19,7 @@ package com.github.blindpirate.gogradle.core.dependency.produce.external.dep
 
 import com.github.blindpirate.gogradle.GogradleRunner
 import com.github.blindpirate.gogradle.core.dependency.produce.external.AbstractExternalDependencyFactoryTest
+import com.github.blindpirate.gogradle.core.pack.GithubGitlabPackagePathResolver
 import com.github.blindpirate.gogradle.support.WithResource
 import com.github.blindpirate.gogradle.util.IOUtils
 import org.junit.Test
@@ -28,7 +29,7 @@ import org.mockito.InjectMocks
 @RunWith(GogradleRunner)
 class DepDependencyFactoryTest extends AbstractExternalDependencyFactoryTest {
     @InjectMocks
-    DepDependencyFactory depDependencyFactory
+    DepDependencyFactory depDependencyFactory = new DepDependencyFactory(new GithubGitlabPackagePathResolver('github.com'))
 
     @Test
     @WithResource('')
@@ -47,7 +48,9 @@ class DepDependencyFactoryTest extends AbstractExternalDependencyFactoryTest {
         verifyMapParsed([name       : "github.com/Masterminds/vcs",
                          subpackages: ['...'],
                          commit     : '3084677c2c188840777bff30054f2b553729d329',
-                         tag        : "v1.11.1"])
+                         tag        : "v1.11.1",
+                         vcs        : "git",
+                         urls       : ['https://github.com/Masterminds/vcs.git', 'git@github.com:Masterminds/vcs.git']])
         verifyMapParsed([name       : "github.com/armon/go-radix",
                          branch     : 'master',
                          subpackages: ['...'],
@@ -73,6 +76,7 @@ class DepDependencyFactoryTest extends AbstractExternalDependencyFactoryTest {
   packages = ["."]
   revision = "3084677c2c188840777bff30054f2b553729d329"
   version = "v1.11.1"
+  source = "github.com/Masterminds/vcs"
 
 [[projects]]
   branch = "master"

--- a/src/test/groovy/com/github/blindpirate/gogradle/core/pack/DefaultPackagePathResolverTest.groovy
+++ b/src/test/groovy/com/github/blindpirate/gogradle/core/pack/DefaultPackagePathResolverTest.groovy
@@ -52,7 +52,7 @@ class DefaultPackagePathResolverTest {
 
     @Before
     void setUp() {
-        resolver = new DefaultPackagePathResolver([resolver1, resolver2])
+        resolver = new DefaultPackagePathResolver(resolver1, resolver2)
         when(resolver1.produce(packagePath)).thenReturn(empty())
         when(resolver2.produce(packagePath)).thenReturn(of(packageInfo))
     }

